### PR TITLE
radiation absorbtion device mutual exclusivity

### PR DESCRIPTION
**Pull request by CrazyAmphibian**
[tweak]
## What this does
makes it so that both versions of the radiation absorption device cannot be added if the other version is present

## Why it's good
having both modules on the same suit leads to really buggy interactions. chose the approach of disallowing both existing after a small vote on the thread.

## Changelog
:cl:
 * rscdel: the radiation absorption device cannot be added to a suit with the advanced version and vice versa

### CODE REVIEW

**Comment by DorainGrey** [comment]
Using both of these was the only way for an Engineer to handle a crystal in space without dying almost instantly to radiation, or enter a crystal/improved shard chamber to prevent a delamination.
You can't wear bio-suits, without pressure killing you. (These also only give something like 75% radiation protection).

Strongly suggest you buff the improved module, if you are going down this road.

**Comment by pedr69** [comment]
>buff the improved module
>more roundstart CE buffs

-1 from me
the radsuits with 100% resist are right outside the SM chamber

i do think there should be a way to get space+rad resist but not with the hardsuit upgrader, i hate the powergay-maxx rigsuits with a passion
maybe make `/datum/reagent/lithotorcrazine` not shit?

**Comment by CrazyAmphibian** [comment]
> Using both of these was the only way for an Engineer to handle a crystal in space without dying almost instantly to radiation

that's not true and the code shows it.
the modules do not stack their absorbancy. as soon as the first module runs out of juice, you lose your immunity regardless if another one exists.

**Comment by CrazyAmphibian** [comment]
also how do i make it so that the commits of previous PRs aren't shown on this PR?
i tried making another branch but it for some reason it didn't let me
is github retarded or am i?

**Comment by pedr69** [comment]
make the branch when you commit the first change (assuming you're browsercoding)
then all the other commits you add to that new branch instead of bleeding edge

**Comment by CrazyAmphibian** [comment]
sorry i meant fork, not branch

**Comment by Hinaichigo** [comment]
Github only allows one fork of a given repo per account. But you can make forks (copies of the repo) and stuff freely on desktop.
I think usually those problems with old commits showing up happen when you make new branches off another branch instead of branching off the "clean" Bleeding-Edge branch. The Bleeding-Edge of your fork should be a replica of the upstream's Bleeding-Edge (vgstation-coders/vgstation13), and should be what you make all new branches from. (so always change back to Bleeding-Edge before making a new branch).
To fix it you have to do something like this:
https://github.com/vgstation-coders/vgstation13/pull/32184#issuecomment-1062615895
and there's another similar way I think. But either way usually ends with "force pushing"

**Comment by CrazyAmphibian** [comment]
> To fix it you have to do something like this: [#32184 (comment)](https://github.com/vgstation-coders/vgstation13/pull/32184#issuecomment-1062615895) and there's another similar way I think. But either way usually ends with "force pushing"

alright, once the PR gets merged/closed i'll try to see if i can't do that

**Comment by Hinaichigo** [comment]
After the PR gets closed/merged you don't have to do that stuff since that's just for removing the old commits from showing in this PR. Instead after this you should make sure your Bleeding-Edge is a mirror of the upstream Bleeding-Edge by either:

- browser: deleting your fork and re-forking a fresh copy
- desktop: [do this](https://stackoverflow.com/questions/1628088/reset-local-repository-branch-to-be-just-like-remote-repository-head) 

and then in the future make new branches for each PR

**Comment by DorainGrey** [comment]
> > Using both of these was the only way for an Engineer to handle a crystal in space without dying almost instantly to radiation
> 
> that's not true and the code shows it. the modules do not stack their absorbancy. as soon as the first module runs out of juice, you lose your immunity regardless if another one exists.

That is not how it works.
It's actually bugged so that you have infinite radiation protection, which is good that you are fixing however I think you'll find that the protection will be extremely limited now.

**Comment by DorainGrey** [comment]
> > buff the improved module
> > more roundstart CE buffs
> 
> -1 from me the radsuits with 100% resist are right outside the SM chamber

The radsuits are not 100% radiation resistant for a start, moreover the CE like normal Engineers starts with the standard radiation absorption device. So it has nothing to do with that...

**Comment by DorainGrey** [comment]
> that's not true and the code shows it.

Also this is true, it's basically the only way to handle a normal crystal in a chamber without dying in 10 seconds.
Whether it justifies it or not, this is easily demonstratable.

**Comment by pedr69** [comment]
yes, i'm not talking about the engi hardsuit but the radiation suit
yknow, the thing in the radiation lockers that comes with geiger counters?
image

https://github.com/vgstation-coders/vgstation13/blob/7029e794f088d6cb70699466442d7be0279ea0cb/code/modules/clothing/suits/utility.dm#L160

**Comment by DorainGrey** [comment]
> yes, i'm not talking about the engi hardsuit but the radiation suit yknow, the thing in the radiation lockers that comes with geiger counters? image
> 
> https://github.com/vgstation-coders/vgstation13/blob/7029e794f088d6cb70699466442d7be0279ea0cb/code/modules/clothing/suits/utility.dm#L160

Yes, I'm well aware of them, especially the fact they aren't really useful at all because they provide no heat or cold protection. Meaning they are only really good for watching, not doing anything at all in a shard chamber or anything similar.

**Comment by CrazyAmphibian** [comment]
What kind of setup are you running that warrants entering the chamber while it's on?
Literally just turn it off and wait 20 seconds for the bulk of the radiation to dissipate.

You shouldn't *need* that much radiation immunity

**Comment by DorainGrey** [comment]
> What kind of setup are you running that warrants entering the chamber while it's on? Literally just turn it off and wait 20 seconds for the bulk of the radiation to dissipate.
> 
> You shouldn't _need_ that much radiation immunity

I'm thinking about any kind of delamination, usually AI/Pulse Demon doing it, very handy to not drop dead entering a chamber.
Whatever though, it's all theorycrafting but it's kind of annoying people pretending this doesn't make a difference.

**Comment by CrazyAmphibian** [comment]
> I'm thinking about any kind of delamination, usually AI/Pulse Demon doing it, very handy to not drop dead entering a chamber. Whatever though, it's all theorycrafting but it's kind of annoying people pretending this doesn't make a difference.

it really doesn't. if for whatever reason you need to go into the chamber RIGHT NOW, you can just use the radsuits. sure, you'll take some pressure/temp damage, but not enough to kill you if you're being fast about extracting the shard into space.
and there's more ways than going into the chamber to save a delaminating shard.

it's called a tradeoff. If hardsuits were able to get 100% uptime on full rad immunity there'd be no point of the radsuits being in the game.

**Comment by falcon2346** [comment]
> I'm thinking about any kind of delamination, usually AI/Pulse Demon doing it, very handy to not drop dead entering a chamber. Whatever though, it's all theorycrafting but it's kind of annoying people pretending this doesn't make a difference.

In a crisis situation why wouldn't you just deconstruct the floor? It's not hard to do that safely with a rad suit on if you have an air shield or hell even just inflatable doors in your set up. You should have those anyways if you're working with something that's going to instantly eat through your rad modules like an active crystal, that's just basic safety.

File: code/modules/clothing/spacesuits/rig_modules/rig_modules.dm [hunk 1]
```diff
@@ -299,7 +299,23 @@
 	if(current_capacity >= max_capacity)
 		deactivate()
 
+/obj/item/rig_module/rad_shield/can_install(var/obj/item/clothing/suit/space/rig/target)
+   if(!..())
+      return FALSE
+   if(locate(/obj/item/rig_module/rad_shield/adv) in target.modules) //don't allow both rad mods at once
```
Comment (Kurfursten) [inline code/modules/clothing/spacesuits/rig_modules/rig_modules.dm:305]: it's very strange that this is necessary... this should be handled by the super call; locate(type) should find children; have you tested this?

Comment (CrazyAmphibian) [reply code/modules/clothing/spacesuits/rig_modules/rig_modules.dm:305]: i don't know what you mean by that. what is the super call?
all i know is that the previous locate check doesn't consider descendant objects to be the parent. If it's supposed to, then it's not.
as for tested, yes, i tested it in a local game, trying to install both modules onto each other, with this implementation successfully blocking that happening.

**Comment by Kurfursten** [comment]
Very strange that locate() is checking for an exact match; I tested it myself and reproduced the bug. I was sure locate didn't work like that.

**Pull request merged**